libstorage/BasicRocksDB.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * @CopyRight:
+ * FISCO-BCOS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2019 fisco-dev contributors.
+ */
+/**
+ * @brief : implement basic interface to access RocksDB
+ * @author: yujiechen
+ * @date: 2019-06-22
+ */
+
+#include "BasicRocksDB.h"
+#include <libdevcore/Exceptions.h>
+using namespace dev;
+using namespace rocksdb;
+
+namespace dev
+{
+namespace db
+{
+/**
+ * @brief: open rocksDB
+ *
+ * @param options: options used to open the rocksDB
+ * @param dbname: db name
+ * @return std::shared_ptr<rocksdb::DB>:
+ * 1. open successfully: return the DB handler
+ * 2. open failed: throw exception(OpenDBFailed)
+ */
+std::shared_ptr<rocksdb::DB> BasicRocksDB::Open(const Options& options, const std::string& dbname)
+{
+    ROCKSDB_LOG(INFO) << LOG_DESC("open rocksDB handler");
+    DB* db = nullptr;
+    auto status = DB::Open(options, dbname, &db);
+    if (!status.ok() || !db)
+    {
+        std::stringstream exitInfo;
+        exitInfo << "Database open error, dbname = " << dbname
+                 << ", status = " << status.ToString();
+        ROCKSDB_LOG(ERROR) << LOG_DESC(exitInfo.str());
+        errorExit(exitInfo, OpenDBFailed());
+    }
+    m_db.reset(db);
+    return m_db;
+}
+
+Status BasicRocksDB::Get(ReadOptions const& options, std::string const& key, std::string& value)
+{
+    assert(m_db);
+    auto status = m_db->Get(options, Slice(std::move(key)), &value);
+    if (!status.ok() && !status.IsNotFound())
+    {
+        std::stringstream exitInfo;
+        exitInfo << "Query rocksDB failed, status:" << status.ToString();
+        ROCKSDB_LOG(ERROR) << LOG_DESC(exitInfo.str());
+        errorExit(exitInfo, QueryDBFailed());
```

Comment:
it will print error log twice here

Reply:
This has been replaced with checkStatus.